components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java

```diff
@@ -1151,7 +1152,13 @@ public static String replaceUrlOriginWithPlaceholders(String absoluteUrl, String
         if (StringUtils.isEmpty(basePath)) {
             return replaceUrlOriginWithPlaceholders(absoluteUrl);
         }
-        return StringUtils.replace(absoluteUrl, basePath, BASE_URL_PLACEHOLDER);
+        absoluteUrl = StringUtils.replace(absoluteUrl, basePath, BASE_URL_PLACEHOLDER);
+
+        if (ApplicationConstants.MY_ACCOUNT_APPLICATION_NAME.equals(appName)) {
+            String consoleBasePath = IdentityUtil.getProperty(CONSOLE_ACCESS_ORIGIN);
+            absoluteUrl = StringUtils.replace(absoluteUrl, consoleBasePath, BASE_URL_CUSTOM_PLACEHOLDER);
+        }
```

Comment:
Why exactly we need this change?

Reply:
We need to create MY_ACCOUNT/Impersonator application role only after creation of my account app.